### PR TITLE
Login-serv initial c++ version.

### PR DESCRIPTION
**Pull request by lighta**
ref : #2478

* **Addressed Issue(s)**: 

* **Server Mode**: 

* **Description of Pull Request**:

### CODE REVIEW

File: src/login/account.cpp [hunk 1]
```diff
@@ -517,25 +517,25 @@ static bool mmo_auth_fromsql(AccountDB_SQL* db, struct mmo_account* acc, uint32
...
-	Sql_GetData(sql_handle,  7, &data, NULL); acc->unban_time = atol(data);
-	Sql_GetData(sql_handle,  8, &data, NULL); acc->expiration_time = atol(data);
-	Sql_GetData(sql_handle,  9, &data, NULL); acc->logincount = (unsigned int) strtoul(data, NULL, 10);
-	Sql_GetData(sql_handle, 10, &data, NULL); safestrncpy(acc->lastlogin, data==NULL?"":data, sizeof(acc->lastlogin));
-	Sql_GetData(sql_handle, 11, &data, NULL); safestrncpy(acc->last_ip, data, sizeof(acc->last_ip));
-	Sql_GetData(sql_handle, 12, &data, NULL); safestrncpy(acc->birthdate, data==NULL?"":data, sizeof(acc->birthdate));
-	Sql_GetData(sql_handle, 13, &data, NULL); acc->char_slots = (uint8) atoi(data);
-	Sql_GetData(sql_handle, 14, &data, NULL); safestrncpy(acc->pincode, data, sizeof(acc->pincode));
-	Sql_GetData(sql_handle, 15, &data, NULL); acc->pincode_change = atol(data);
+	Sql_GetData(sql_handle,  0, &data, &out_len); acc->account_id = atoi(data);
```
Comment (Lemongrass3110) [inline src/login/account.cpp]: This results in 17 useless operations, since the value is not used.

File: src/login/logincnslif.hpp [hunk 2]
```diff
@@ -7,13 +7,27 @@
  * @author rAthena Dev Team
  */
 
+#pragma once
 #ifndef CONSOLEIF_H
 #define	CONSOLEIF_H
 
 #ifdef	__cplusplus
```
Comment (Lemongrass3110) [inline src/login/logincnslif.hpp]: Shouldnt this simpl ybe removed?

File: src/login/CMakeLists.txt [hunk 3]
```diff
@@ -7,8 +7,8 @@ set( LOGIN_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}  CACHE INTERNAL "" )
 #
 if( BUILD_SERVERS )
 message( STATUS "Creating target login-server" )
-file(GLOB LOGIN_HEADERS ${LOGIN_SOURCE_DIR}/*.h)
-file(GLOB LOGIN_SOURCES_C ${LOGIN_SOURCE_DIR}/*.c)
+file(GLOB LOGIN_HEADERS ${LOGIN_SOURCE_DIR}/*.hpp)
+#file(GLOB LOGIN_SOURCES_C ${LOGIN_SOURCE_DIR}/*.c)
```
Comment (Lemongrass3110) [inline src/login/CMakeLists.txt]: I think we could remove it completely. Guess we will not need it again.

File: src/login/account.cpp (hunk 1 above)
Comment (aleos89) [reply src/login/account.cpp]: Agreed. This wasn't done in other cases, especially for the char-server. Is there a reason it needs to be done here?

Comment (lighta) [reply src/login/account.cpp]: only for string, but I didn't change those yet.
I'musing the std::string(const char*, size_t) constructor.

Comment (Lemongrass3110) [reply src/login/account.cpp]: Then I'd love to revert it as long as we dont use std::string. Might be overkill to do this in a preparation pull though.

**Pull request merged**